components/eventing-controller/options/options.go

```diff
@@ -0,0 +1,63 @@
+package options
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+)
+
+const (
+	// args
+	argNameMaxReconnects   = "max-reconnects"
+	argNameMetricsAddr     = "metrics-addr"
+	argNameReconnectWait   = "reconnect-wait"
+	argNameReconcilePeriod = "reconcile-period"
+
+	// env
+	envNameLogFormat = "APP_LOG_FORMAT"
+	envNameLogLevel  = "APP_LOG_LEVEL"
+)
+
+type Options struct {
```

Comment:
I like the idea of an own options package a lot, like the log package. Only would also add an `options.New()`  here. Especially for once possibly needing executed logic when the options instance is created.

Also would add comments for all exported identifiers.